mmpose/core/evaluation/mesh_eval.py

```diff
@@ -0,0 +1,68 @@
+# ------------------------------------------------------------------------------
+# Adapted from https://github.com/nkolot/SPIN
+# Original licence: Copyright (c) 2018 akanazawa, under the MIT License.
+# ------------------------------------------------------------------------------
+
+import numpy as np
+
+
+def compute_similarity_transform(source_points, target_points):
+    """Computes a similarity transform (sR, t) that takes a set of 3D points
+    source_points (N x 3) closest to a set of 3D points target_points, where R
+    is an 3x3 rotation matrix, t 3x1 translation, s scale. And return the
+    transformed 3D points source_points_hat (N x 3). i.e. solves the orthogonal
+    Procrutes problem.
+
+    Notes:
+        Points number: N
+
+    Args:
+        source_points (np.ndarray([N, 3])): Source point set.
+        target_points (np.ndarray([N, 3])): Target point set.
+
+    Returns:
+        source_points_hat (np.ndarray([N, 3])): Transformed source point set.
+    """
+
+    transposed = False
+    if source_points.shape[0] != 3 and source_points.shape[0] != 2:
+        source_points = source_points.T
+        target_points = target_points.T
+        transposed = True
+    assert (target_points.shape[1] == source_points.shape[1])
+
+    # 1. Remove mean.
+    mu1 = source_points.mean(axis=1, keepdims=True)
```

Comment:
double check that this is the average of three numbers `(x, y, z)` for each point

Reply:
i guess, either the ` source_points (np.ndarray([N, 3]))` is not true, or the code is wrong

Reply:
okay it transposes the data in the beginning